src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAISerializationHelpers.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.ServerSentEvents;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using OpenAI.Chat;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Defines a set of helpers used to serialize Microsoft.Extensions.AI content using the OpenAI wire format.
+/// </summary>
+public static class OpenAISerializationHelpers
```

Comment:
Seems reasonable. I don't have anything significantly better.

Reply:
Actually I should have mentioned before I marked 'approve' -

I do wonder whether this functionality should all really be in the same package as the client library. Presumably 99.9% of people will be using MEAI.OpenAI to act as a client, so exposing all the opposite-direction APIs for those people may be confusing, or even limiting for us in the future if we want to reference server-specific helpers but can't do so because we don't want extra package dependencies.

Is there any major payoff from this all being in a single MEAI.OpenAI package? I guess it makes it easier to use `internal` helpers but we could IVT for that if needed.

Reply:
It's something we could consider assuming the server-side components do accrue additional dependencies (for example when we replace the polyfill with the S.N.SSE package after next November). I think it's a bit premature to do so at this stage -- I don't believe the presence of a helper class is enough to put people off about the purpose of the library.

Reply:
Thinking forward to when the abstraction implementations are in the OpenAI library itself, what would we want the distribution vehicle of these additional helpers to be?

Reply:
I'm raising it because it would be a breaking change to lift these pieces into another package, so if we think it's likely we want to depend on the SSE package within the forseeable future, we could save ourselves the future trouble of dealing with a breaking change.

Reply:
> Thinking forward to when the abstraction implementations are in the OpenAI library itself, what would we want the distribution vehicle of these additional helpers to be?

My expectation is that an official OpenAI implementation would bundle both pieces of functionality in the same package. I don't believe the SSE dependency is going to be a huge differentiator -- they already bundle some sort of SSE implementation for the parsing side of things.

Reply:
It sounds like you're saying yes, there will be a breaking change in the future, but it will be to deprecate MEAI.OpenAI entirely and migrate it into the OpenAI client package. Is that right? If so then I agree we don't really benefit from splitting MEAI.OpenAI at this stage.

Reply:
I think that's our ideal endgame. Keeping the mappers in sync and up to date with newer versions of the OpenAI library isn't sustainable longer term.